great_expectations/experimental/metric_repository/metric_list_retriever.py

```diff
@@ -0,0 +1,322 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Hashable
+from itertools import chain
+from typing import TYPE_CHECKING, Any, List, Optional, Sequence, Tuple, Union
+
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.experimental.metric_repository.metric_retriever import (
+    MetricRetriever,
+)
+from great_expectations.experimental.metric_repository.metrics import (
+    ColumnMetric,
+    Metric,
+    MetricTypes,
+    TableMetric,
+)
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.datasource.fluent.batch_request import BatchRequest
+    from great_expectations.validator.validator import Validator
+
+logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
+
+_MetricKey: TypeAlias = Union[Tuple[str, Hashable, Hashable], Tuple[str, str, str]]
+
+
+class MetricListMetricRetriever(MetricRetriever):
+    def __init__(self, context: AbstractDataContext):
+        super().__init__(context=context)
+        self._validator: Validator | None = None
+
+    @override
+    def get_metrics(
+        self,
+        batch_request: BatchRequest,
+        metric_list: Optional[List[MetricTypes]] = None,
+    ) -> Sequence[Metric]:
+        metrics_result: List[Metric] = []
```

Comment:
Nit - you could consider moving this closer to where it is first used, that way it's not allocated if the method exits early (e.g. raising if the metric_list is empty).